Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -1297,6 +1308,51 @@ public extension Purchases {
     }
 }
 
+// MARK: - Virtual Currencies
+#if !ENABLE_CUSTOM_ENTITLEMENT_COMPUTATION
+public extension Purchases {
+
+    @objc func virtualCurrencies(
+        completion: @escaping @Sendable (VirtualCurrencies?, PublicError?) -> Void
+    ) {
+        OperationDispatcher.dispatchOnMainActor {
+            Task {
+                do {
+                    let virtualCurrencies = try await self.virtualCurrencies()
+                    completion(virtualCurrencies, nil)
```

Comment:
I think we should remove the outer `dispatchOnMainActor` and wrap only the completion calls in it. For reference check how we're doing it in `eligibleWinBackOffers`